### DISCUSSION TITLE
running command based on docker output

### DISCUSSION THREAD

**Discussion by Ahptt** [itzg/docker-minecraft-server#2049] [Q&A] [answered]
I run a 1.19 based server with the autopause feature enabled. To my understanding and testing, I can't connect to the server console via rcon. My situation is I am mining monero on the same system that I'm hosting a server. I want to be able to trigger my mining to stop when any client joins the server, and start up again once all clients leave.

The only idea I can come up with is a script that checks every 5 or 10 minutes the most recent log from `docker logs ___ | grep joined`, and also `docker logs ___ | grep All\ clients\ disconnected` and then run and if else to produce the desired result.

This seems very inefficient to me as this will have to run every so often, and I wonder with my current setup if there is a better way to ultimately trigger this, instead of consistently checking. Please let me know if you have any ideas or have done something similar. It's much appreciated! :smile:

**Comment by itzg** [comment]
I actually like your idea of the periodic query and would suggest trying out a semi-builtin approach like this snippet used by auto-pause:

https://github.com/itzg/docker-minecraft-server/blob/0d5e44a3387bfd5462342e5729351b719c357178/files/auto/autopause-fcns.sh#L23-L33

**Reply by Ahptt** [reply]
That looks great! My only question is how do I add my configuration to the container? Is this something I would modify and then build a container out of, or can I pass it through docker-compose as I would mods/datapacks? Thank you.

**Reply by itzg** [reply, marked as answer]
Good question. I would suggest building a separate image to run as a sidecar where you could pull in the mc-monitor binary from https://github.com/itzg/mc-monitor/releases/latest

With that, have your entrypoint be a script of some form that runs in a loop checking the status and taking actions needed.
